src/downloadManager/__init__.py

```diff
@@ -155,6 +160,7 @@ def finishedDownload(self, dler):
 
         if not dler.succeeded():
             dler.dest.remove()
+            os.unlink(filepath)
```

Comment:
We should return at the end in this branch to avoid updating widgets with an icon that doesn't exist. I got an exception when quitting a coop game host widget.